File: algorithms/Codility-master/Codility-master/2_Arrays/oddOccurencesInArray.js



function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    // 100% https://app.codility.com/demo/results/training2KHES6-2P7/
    if (A.length == 1){
        return A[0]
    }

    var dict = new Object();
    for (let i=0 ; i < A.length ; i++) {
        if (A[i] in dict){
            dict[A[i]] += 1
        }
        else{
            dict[A[i]] = 1
        }
    }
    for (var item in dict) {
        if(dict[item] % 2 != 0){
            return Number(item)
        }
    }
}

File: algorithms/Codility-master/Codility-master/2_Arrays/oddOccurencesInArray.py

# 66% 
def solutionB(A):
    
    if len(A) == 1:
        return A[0]
    # write your code in Python 3.6
    nums = []
    for num in A:
        if num in nums:
            del nums[nums.index(num)]
        else:
             nums.append(num)
            
    return nums[0]

# 100%
import collections
def solution(A):
    
    if len(A) == 1:
        return A[0]
        
    c = collections.Counter(A)
    for item in c:
        if (c[item] % 2) != 0:
            return item
